src/main/java/com/rocketden/main/service/GameManagementService.java

```diff
@@ -95,6 +99,33 @@ public RoomDto startGame(String roomId, StartGameRequest request) {
         return roomDto;
     }
 
+    public RoomDto playAgain(String roomId, PlayAgainRequest request) {
+        Game game = getGameFromRoomId(roomId);
+
+        if (game.getGameTimer() == null || !game.getGameTimer().isTimeUp()) {
+            throw new ApiException(GameError.GAME_NOT_OVER);
+        }
+
+        Room room = game.getRoom();
+        User initiator = UserMapper.toEntity(request.getInitiator());
+        if (!room.getHost().equals(initiator)) {
+            throw new ApiException(GameError.INVALID_PERMISSIONS);
+        }
+
+        // Set all users to be disconnected
+        room.getUsers().forEach((user) -> user.setSessionId(null));
```

Comment:
Does this actually disconnect the users on the frontend? If not, what is this used for?

Reply:
This is basically to ensure the list of active and inactive users are correct and re-trigger the WebSocketConnectionEvents code - without it, I think everyone would show up to the lobby page as connected even if they've already closed the browser